File: packages/typescript/src/roles/tech-lead.ts
import { Agent } from "../agent.js";
import { AgentRole, TaskType } from "../types.js";
import type { AgentConfig } from "../types.js";

const HANDLED_TASK_TYPES: TaskType[] = [
  TaskType.BackendFeature,
  TaskType.FrontendFeature,
  TaskType.CodeReview,
  TaskType.SprintPlanning,
  TaskType.Documentation,
];

export class TechLead extends Agent {
  constructor(config: Partial<AgentConfig> = {}) {
    super({
      role: AgentRole.TechLead,
      ...config,
    });
  }

  canHandle(taskType: TaskType): boolean {
    return HANDLED_TASK_TYPES.includes(taskType);
  }
}

File: packages/python/src/tcs_bpi_ai_team/roles/tech_lead.py
"""Tech Lead agent."""

from __future__ import annotations

from tcs_bpi_ai_team.agent import Agent
from tcs_bpi_ai_team.types import AgentConfig, AgentRole, TaskType

_HANDLED_TASK_TYPES = [
    TaskType.BACKEND_FEATURE,
    TaskType.FRONTEND_FEATURE,
    TaskType.CODE_REVIEW,
    TaskType.SPRINT_PLANNING,
    TaskType.DOCUMENTATION,
]


class TechLead(Agent):
    """Tech Lead / Architect — architecture decisions, code review, technical direction."""

    def __init__(self, config: AgentConfig | None = None) -> None:
        super().__init__(config or AgentConfig(role=AgentRole.TECH_LEAD))

    def can_handle(self, task_type: TaskType) -> bool:
        return task_type in _HANDLED_TASK_TYPES
